crates/iceberg/src/scan/context.rs

```diff
@@ -186,16 +187,25 @@ impl PlanContext {
         tx_data: Sender<ManifestEntryContext>,
         delete_file_idx: DeleteFileIndex,
         delete_file_tx: Sender<ManifestEntryContext>,
-    ) -> Result<Box<impl Iterator<Item = Result<ManifestFileContext>> + 'static>> {
+    ) -> Result<(Vec<Result<ManifestFileContext>>, ManifestMetrics)> {
```

Comment:
Since we were returning a vector and this function was only called at a single place, I took the liberty of changing the return value. This somewhat simplified passing the result to  a spawned thread because `Vec` implies `Send + Sync` when it's Items do.

I've also extended the TODO comment below for future reference because I've added another obstacle to simply using an iterator here: the `ManifestMetrics` are now continuously mutated in the loop. If we used an iterator instead, we couldn't as easily (I think) pass around the mutable reference.